### ISSUE TITLE
Configuration to get media folder from Flysystem location

### ISSUE THREAD

**Issue by BrocksiNet** [Flagbit/Magento2-Flysystem#23]
The idea is to have a backend config where you can tell magento to use flysystem as media storage instead of the local path.

Use case would be for example: A developer team has all the product images from a customer in a S3. Instead of that everybody is downloading the images you just change the config and the frontend will get the images from the s3 instead from local.

**Comment by robin-homberg** [comment]
Thank you Björn. But here some stuff to think about:

1. This would be an awesome feature to be used together with S3. But how to handle this for other Adapters like SFTP ?
An answer could be to always download the file to the tmp folders in magento, if trying to implement them, but when integrating this it should be taken care about other Flysystem adapters, where you cant access the file directly

2. If you dont want to hook or plugin a whole lot of core functionality it will be pretty tough to deal with the image paths for product and category images. Some ideas to handle this:
- Define a new product image type like "flysystem_image" to handle the new functionality
- synchronize the media folder of magento with the flysystem file storage to provide the same relative paths on both sides

**Comment by BrocksiNet** [comment]
Good point (**1**). For me it would also be okay to move this feature to the S3 repository.
